inc/class-upgrade-history.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internal
+ */
+
+/**
+ * This class handles storing the current options for future reference.
+ *
+ * This should only be used during an upgrade routine.
+ */
+class WPSEO_Upgrade_History {
+
+	/**
+	 * Option to use to store/retrieve data from.
+	 *
+	 * @var string
+	 */
+	protected $option_name = 'wpseo_upgrade_history';
+
+	/**
+	 * WPSEO_Upgrade_History constructor.
+	 *
+	 * @param null|string $option_name Optional. Custom option to use to store/retrieve history from.
+	 */
+	public function __construct( $option_name = null ) {
```

Comment:
Is it necessary to support setting that value? I'd assume it will always be 'wpseo_upgrade_history', and there'd be no point in changing it. In that case, we could remove the constructor parameter and use a `const` instead of a property for it. Correct me if I'm wrong in my assumption though.